diff --git a/build.zig b/build.zig
index 53ce90f..bf6df2a 100644
--- a/build.zig
+++ b/build.zig
@@ -204,8 +204,8 @@ fn benchTargets(
     var map = std.StringHashMap(*std.Build.Step.Compile).init(b.allocator);
 
     // Open the directory
-    const c_dir_path = "src/bench";
-    var c_dir = try std.fs.cwd().openDir(comptime thisDir() ++ "/" ++ c_dir_path, .{ .iterate = true });
+    const c_dir_path = b.path("src/bench").getPath(b);
+    var c_dir = try std.fs.cwd().openDir(c_dir_path, .{ .iterate = true });
     defer c_dir.close();
 
     // Go through and add each as a step
@@ -218,7 +218,7 @@ fn benchTargets(
         // Name of the app and full path to the entrypoint.
         const name = entry.name[0..index];
         const path = try std.fs.path.join(b.allocator, &[_][]const u8{
-            c_dir_path,
+            "src/bench",
             entry.name,
         });
 
@@ -261,7 +261,7 @@ fn exampleTargets(
     if (!install) return;
 
     // Open the directory
-    const c_dir_path = (comptime thisDir()) ++ "/examples";
+    const c_dir_path = b.path("examples").getPath(b);
     var c_dir = try std.fs.cwd().openDir(c_dir_path, .{ .iterate = true });
     defer c_dir.close();
 
@@ -280,7 +280,7 @@ fn exampleTargets(
         // Name of the app and full path to the entrypoint.
         const name = entry.name[0..index];
         const path = try std.fs.path.join(b.allocator, &[_][]const u8{
-            c_dir_path,
+            "examples",
             entry.name,
         });
 
@@ -341,8 +341,3 @@ fn exampleTargets(
         }
     }
 }
-
-/// Path to the directory with the build.zig.
-fn thisDir() []const u8 {
-    return std.fs.path.dirname(@src().file) orelse unreachable;
-}